python-avd/pyavd/_eos_designs/structured_config/mlag/__init__.py

```diff
@@ -377,6 +377,29 @@ def _router_bgp_mlag_peer_group(self) -> dict:
 
         router_bgp["peer_groups"] = [strip_empties_from_dict(peer_group)]
 
+        # If self.inputs.mlag_ipv4_vrfs_peer is defined in the input
+        if (
+            self.inputs.bgp_peer_groups.mlag_ipv4_vrfs_peer
+            and self.inputs.bgp_peer_groups.mlag_ipv4_vrfs_peer.name != self.inputs.bgp_peer_groups.mlag_ipv4_underlay_peer.name
+        ):
```

Comment:
would rather that we don't have to do this again here - maybe lets make it a boolean in shared_utils still.

So i. n shared_utils 2 properties: a boolean to decide if this should be render + what you have today returning the correct peer_group structured to be used in mlag and in network_Services

Reply:
so after further discussion:

create two shared_utils:

1. a boolean doing your check - `use_separate_peer_group_for_mlag_vrfs`
2. return the name - `mlag_vrfs_peer_group_name`

Use 1 in mlag __init__ to decide if the peer group should be rendered and then pick the data from inputs
Use 2 in network_services.router_bgp

Reply:
Done!